### PR TITLE
Introduce fit_bounds method

### PR DESCRIPTION
**Pull request by themiurgo**
This pull request introduces the possibility to fit the map to a bounding box and solves issue #118.

### CODE REVIEW

**Comment by ocefpaf** [comment]
Thanks @themiurgo!  The PR looks good.

Can you update the [change log](https://github.com/python-visualization/folium/blob/master/CHANGES.txt) and squash the commits?

**Comment by themiurgo** [comment]
Done.

**Comment by ocefpaf** [comment]
Thanks!  I have two more requests :wink: (If you are up to them, otherwise I will merge as is and we can open issues to address them later):

1) Add a small test to your template file.  (See an example [here](https://github.com/python-visualization/folium/blob/master/tests/folium_tests.py#L106--117).)

2) Can you go the extra mile and implement the padding options?
http://leafletjs.com/reference.html#map-paddingtopleft

Ciao!

**Comment by themiurgo** [comment]
Sure, I love folium, so I'm happy to contribute. I'll get to it.

**Comment by themiurgo** [comment]
Will you ask me to squash the new commits as well? :stuck_out_tongue:

**Comment by ocefpaf** [comment]
Not right now.  I commit as freely as you want, and lets squash once we are ready to merge.  Is that OK for you?

**Comment by themiurgo** [comment]
Ok, let me know if this looks good.

File: tests/folium_tests.py [hunk 1]
```diff
@@ -440,3 +440,28 @@ def test_multi_polyline(self):
...
+        bounds = ((52.193636, -2.221575), (52.636878, -1.139759))
+        fit_bounds_tpl = self.env.get_template('fit_bounds.js')
+        fit_bounds_rendered = fit_bounds_tpl.render({
+           'bounds': json.dumps(bounds),
+           'fit_bounds_options': {},
+        })
+
+        self.map.fit_bounds(bounds)#, max_zoom=15, padding=(3, 3))
+        print self.map.template_vars['fit_bounds']
+        print fit_bounds_rendered
```
Comment (ocefpaf) [inline tests/folium_tests.py]: Nice test!  Can you remove the print statements?

File: tests/folium_tests.py [hunk 2]
```diff
@@ -440,3 +440,28 @@ def test_multi_polyline(self):
...
+
+        fit_bounds_tpl = self.env.get_template('fit_bounds.js')
+        fit_bounds_rendered = fit_bounds_tpl.render({
+           'bounds': json.dumps(bounds),
+           'fit_bounds_options': json.dumps({'padding': (3,3), 'maxZoom': 15}),
+        })
+
+        self.map.fit_bounds(bounds, max_zoom=15, padding=(3, 3))
+        print self.map.template_vars['fit_bounds']
+        print fit_bounds_rendered
```
Comment (ocefpaf) [inline tests/folium_tests.py]: Same here.

File: folium/folium.py [hunk 3]
```diff
@@ -587,6 +587,50 @@ def click_for_marker(self, popup=None):
...
+        bounds: list of (latitude, longitude) points
+            Bounding box specified as two points [southwest, northeast]
+        padding_top_left: (x, y) point, default None
+            Padding in the top left corner. Useful if some elements in
+            the corner, such as controls, might obscure objects you're zooming
+            to.
+        padding_bottom_right: (x, y) point, default None
+            Padding in the bottom right corner.
+        padding: (x, y) point, default None
+            Equivalent to seting both top left and bottom right padding to
```
Comment (ocefpaf) [inline folium/folium.py]: Typo: seting -> setting

File: folium/folium.py [hunk 4]
```diff
@@ -587,6 +587,50 @@ def click_for_marker(self, popup=None):
...
+            the same value.
+        max_zoom: int, default None
+            Maximum zoom to be used.
+
+        Example
+        -------
+        >>>map.fit_bounds([[52.193636, -2.221575], [52.636878, -1.139759]])
```
Comment (ocefpaf) [inline folium/folium.py]: Insert a space after `>>>`

**Comment by ocefpaf** [comment]
Looks great!  Just remove the `print` statements so the Python 3 tests will work.

Thanks for going the extra mile!  This is a nice addition to folium.

File: folium/folium.py (hunk 4 above)
Comment (themiurgo) [inline folium/folium.py]: I didn't introduce a space because the rest of the file follows the no-space convention. (I personally prefer the space too)

**Comment by themiurgo** [comment]
Ok, made the edits and squished the changes.

File: folium/folium.py (hunk 4 above)
Comment (ocefpaf) [inline folium/folium.py]: Yep, I want to fix the rest as well!

File: folium/folium.py [hunk 5]
```diff
@@ -587,6 +587,50 @@ def click_for_marker(self, popup=None):
...
+
+        """
+        options = {
+            'paddingTopLeft': padding_top_left,
+            'paddingBottomRight': padding_bottom_right,
+            'padding': padding,
+            'maxZoom': max_zoom,
+        }
+        fit_bounds_options = {}
+        for key, opt in options.iteritems():
```
Comment (ocefpaf) [inline folium/folium.py]: I missed this one but travis caught it!  Just use `items`.  (It will never be such a big iterable to make a difference.)

Comment (themiurgo) [inline folium/folium.py]: Ok, fixed and squashed the commits.

**Comment by ocefpaf** [comment]
Thanks @themiurgo!  We have a merge :tada:

**Comment by ocefpaf** [comment]
Hope to see more of you around here in the future :wink:

**Pull request merged**